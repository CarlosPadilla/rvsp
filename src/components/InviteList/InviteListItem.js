import React from 'react';
import PropTypes from 'prop-types';

const InviteListItem = props => (
  <div>
    <div>{props.invite.id}</div>
    <div>{props.invite.name}</div>
    <div>{props.invite.email}</div>
  </div>
);

InviteListItem.propTypes = {
  invite: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  }).isRequired,
};

export default InviteListItem;
