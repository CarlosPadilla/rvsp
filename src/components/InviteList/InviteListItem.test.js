import React from 'react';
import { shallow } from 'enzyme';
import InviteListItem from './InviteListItem';

describe('InviteListItem Component', () => {

  const invite = {
    id: 1,
    name: 'Aimar',
    email: 'aimar@snapit.ee'
  }

  const app = shallow(<InviteListItem invite={invite} />)

  it('renders', () => {
    expect(app).toMatchSnapshot()
  })

})
