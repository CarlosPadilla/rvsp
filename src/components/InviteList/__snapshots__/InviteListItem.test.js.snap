// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteListItem Component renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InviteListItem
    invite={
        Object {
            "email": "aimar@snapit.ee",
            "id": 1,
            "name": "Aimar",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          1
</div>,
        <div>
          Aimar
</div>,
        <div>
          aimar@snapit.ee
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1,
        },
        "ref": null,
        "rendered": 1,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Aimar",
        },
        "ref": null,
        "rendered": "Aimar",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "aimar@snapit.ee",
        },
        "ref": null,
        "rendered": "aimar@snapit.ee",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            1
</div>,
          <div>
            Aimar
</div>,
          <div>
            aimar@snapit.ee
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1,
          },
          "ref": null,
          "rendered": 1,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Aimar",
          },
          "ref": null,
          "rendered": "Aimar",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "aimar@snapit.ee",
          },
          "ref": null,
          "rendered": "aimar@snapit.ee",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
